lib/cloud_controller/diego/docker/staging_completion_handler.rb

```diff
@@ -36,6 +37,12 @@ def save_staging_result(app, payload)
               end
             end
 
+            if payload.key?(:lifecycle_data) && payload[:lifecycle_data][:docker_image].present?
+              droplet.update_cached_docker_image(payload[:lifecycle_data][:docker_image])
+            else
+              droplet.update_cached_docker_image(nil)
+            end
+
```

Comment:
Maybe consider something like this to avoid duplicating the walk of the keys?

```ruby
            cached_image = payload[:lifecycle_data] && payload[:lifecycle_data][:docker_image]
            droplet.update_cached_docker_image(cached_image) if cached_image.present?
```